src/ast/mod.rs

```diff
@@ -294,8 +294,11 @@ impl fmt::Display for Expr {
                 write!(f, "{}", column)?;
                 for k in keys {
                     match k {
-                        k @ Value::Number(_, _) => write!(f, "[{}]", k)?,
-                        Value::SingleQuotedString(s) => write!(f, "[\"{}\"]", s)?,
+                        Expr::Value(v) => match v {
+                            k @ Value::Number(_, _) => write!(f, "[{}]", k)?,
+                            Value::SingleQuotedString(s) => write!(f, "[\"{}\"]", s)?,
+                            _ => write!(f, "[{}]", k)?,
+                        },
```

Comment:
Do we need to match the types of `key`? Could it all be replated with `write!(f, "[{}], k)`?

Also it would be cool to avoid the duplication. If we need a match, how about

```rust
match k { 
  k@ Expr::Value(Value::Number(_, _) => write!(f, "[{}]", k)?,
  Expr::Value(Value::SingleQuotedString(s)) => write!(f, "[\"{}\"]", s)?,
  _ => write!(f, "[{}]", k)?,
}
```

Reply:
Thank you for your reply, I'll update it.